### PR TITLE
Balloons

### PR DESCRIPTION
**Pull request by Shadowmech88**
bbaalllloonn
Adds balloons. Balloons can be inflated by any mob that breathes or has lungs.
Once inflated, you can tie some cable coil to the balloon, which allows it to be visible when you hold it.
Each balloon is assigned a random color when it spawns.
Balloons can be popped by any pointed object.
Boxes of balloons can be obtained from the party supplies crate orderable from Cargo.

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1086,3 +1087,82 @@
...
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/toys.dmi', "right_hand" = 'icons/mob/in-hand/right/toys.dmi')
+	var/col = "#FFFFFF"
+
+/obj/item/toy/balloon/New(turf/T, var/ccol)
+	..(T)
+	if(ccol)
+		col = ccol
+	else
+		col = rgb(rand(0,255),rand(0,255),rand(0,255))
+	icon += col
```
Comment (PJB3005) [inline code/game/objects/items/toys.dm]: You should really be using the `color` var instead.

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1086,3 +1087,82 @@
...
+
+/obj/item/toy/balloon
+	name = "balloon"
+	desc = "A simple balloon."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "balloon_deflated"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/toys.dmi', "right_hand" = 'icons/mob/in-hand/right/toys.dmi')
+	var/col = "#FFFFFF"
+
+/obj/item/toy/balloon/New(turf/T, var/ccol)
```
Comment (PJB3005) [inline code/game/objects/items/toys.dm]: Make the first argument `atom`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by AI-Is-Rogue** [comment]
Clowns should be able to turn the balloons into various shapes.

**Comment by Intigracy** [comment]
Make the existing glove balloon a subtype of this please.

**Comment by SangriaMonkey** [comment]
>that shading

:grimacing:

**Comment by Shadowmech88** [comment]
If anyone has a better sprite, I'm more than happy to accept it.

@AI-is-rogue
I plan to do that in a later PR.

**Comment by Shadowmech88** [comment]
I could also reparent the Syndicate balloon, but the only functionality it would gain would be popping and I'm not sure if people would want it to be poppable.

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1086,3 +1087,191 @@
...
+	var/col = "#FFFFFF"
+
+/obj/item/toy/balloon/New(atom/A, var/ccol)
+	..(A)
+	if(col)
+		if(ccol)
+			col = ccol
+		else
+			col = rgb(rand(0,255),rand(0,255),rand(0,255))
+		icon += col
```
Comment (PJB3005) [inline code/game/objects/items/toys.dm]: Still need to use `color`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1086,3 +1087,182 @@
...
+	desc = "A simple balloon."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "balloon_deflated"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/toys.dmi', "right_hand" = 'icons/mob/in-hand/right/toys.dmi')
+	w_class = W_CLASS_TINY
+	force = 0
+	throwforce = 0
+	var/col = "#FFFFFF"
+
+/obj/item/toy/balloon/New(atom/A, var/ccol)
```
Comment (ComicIronic) [inline code/game/objects/items/toys.dm]: `ccol` is a pretty bad name, `chosen_col` would be better if that's what you're going for.

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1086,3 +1087,182 @@
...
+
+/obj/item/toy/balloon/glove
+	name = "latex glove"
+	desc = "A latex glove."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "latexballoon"
+	item_state = "lgloves"
+	col = null
+
+/obj/item/toy/balloon/glove/inflate(mob/user, datum/gas_mixture/G)
```
Comment (ComicIronic) [inline code/game/objects/items/toys.dm]: If this is just copy-pasted for the glove spawning, you can make that a type var or a proc.

Comment (Shadowmech88) [reply code/game/objects/items/toys.dm]: How do I reference an object created from a type held in a var? Would
```
var/type = /obj/path
var/O = new type(loc)
```
work?

Comment (ComicIronic) [reply code/game/objects/items/toys.dm]: Yes.

**Comment by gbasood** [comment]
okay but can mommis use these

**Comment by Shadowmech88** [comment]
Yes.

File: code/game/objects/items/toys.dm (hunk 3 above)
Comment (Shadowmech88) [reply code/game/objects/items/toys.dm]: Is not using `color` going to prevent a merge?

**Comment by Blithering** [comment]
Do balloons inflated by plasmamen become tiny bombs?

**Comment by Shadowmech88** [comment]
No, because plasmamen exhale CO2.

**Comment by Blithering** [comment]
Dang. There go my hopes of creepy clown handing out doom balloons becoming a thing

**Comment by Shadowmech88** [comment]
You *can* fill the balloon with some of the contents of an air tank and use that.

**Comment by Shadowmech88** [comment]
New sprite with another attempt at shading.
bbaalllloonn

**Comment by Shadowmech88** [comment]
@PJB3005 
I've fixed it so that it uses the `color` var, now.

**Comment by Probe1** [comment]
Look at you go!  Those look much better.

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -1086,3 +1087,179 @@
...
+	if(user)
+		user.drop_item(src, force_drop = 1)
+		user.put_in_hands(B)
+		to_chat(user, "You blow up \the [src].")
+	playsound(src, 'sound/misc/balloon_inflate.ogg', 50, 1)
+	if(!G)
+		B.air_contents = new /datum/gas_mixture()
+		B.air_contents.volume = 12 //liters
+		B.air_contents.temperature = T20C
+		B.air_contents.carbon_dioxide = 0.5
```
Comment (PJB3005) [inline code/game/objects/items/toys.dm]: Very naive to assume every species breaths out CO2.

Also this should be using adjust to edit the mole counts.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm (hunk 6 above)
Comment (Shadowmech88) [reply code/game/objects/items/toys.dm]: Every species *does* exhale CO2.
But I'll change it just in case that changes in the future.

**Review by PJB3005 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Dang, I forgot the changelog.

**Comment by Probe1** [comment]
Just open a PR of shame

**Pull request merged**